### PR TITLE
Stop setting addons with unlisted versions STATUS_NOMINATED or STATUS_PUBLIC

### PR DESCRIPTION
**Pull request by eviljeff**
Also tidy up editors somewhat.
Part of #3465

### CODE REVIEW

**Comment by diox** [comment]
We do need something to prevent those add-ons from being shown as "Incomplete" in devhub though, right ? Maybe some check on `if addon.status == amo.STATUS_NULL and addon.has_complete_metadata()` ?

**Comment by eviljeff** [comment]
#3471 hides the status indicator for add-ons without listed versions.  So this pull is either blocked on that (there are some outstanding Product/UX questions blocking) or we do a workaround for now that does the minimum to avoid developer confusion.

**Comment by diox** [comment]
Up to you. We need to remember that the same thing needs to be done in both the sidebar and the devhub index that list all your add-ons, though.

**Comment by eviljeff** [comment]
it's the same template snippet 😃

**Comment by eviljeff** [comment]
sidebar done in #4005

File: src/olympia/editors/models.py [hunk 1]
```diff
@@ -257,14 +256,13 @@ def base_query(self):
...
-                'NOT addons.is_listed',
                 'versions.channel = %s' % amo.RELEASE_CHANNEL_UNLISTED,
                 """((reviewed_versions.id = (select max(reviewed_versions.id)))
                     OR
                     (reviewed_versions.id IS NULL))
                 """,
-                'addons.status NOT IN (%s, %s)' % (
-                    amo.STATUS_NULL, amo.STATUS_DISABLED)
+                'addons.status NOT IN (%s)' % (
+                    amo.STATUS_DISABLED)
```
Comment (diox) [inline src/olympia/editors/models.py]: Could replace the `IN` now that there is only one value left.

**Pull request merged**